src/systems/explicit_system.C

```diff
@@ -45,6 +46,8 @@ ExplicitSystem::~ExplicitSystem ()
 {
   // clear data
   this->clear();
+
+  remove_vector("RHS Vector");
```

Comment:
Although, to be fair, that's not obvious.  System::~System() calls this->clear(), and C++ rules say that that means System::clear(), *not* ExplicitSystem::clear(), which in this case is exactly what we want and won't leave a re-added RHS vector, but which is such obscure behavior that CERT recommends [never doing it.](https://www.securecoding.cert.org/confluence/display/cplusplus/OOP50-CPP.+Do+not+invoke+virtual+functions+from+constructors+or+destructors)